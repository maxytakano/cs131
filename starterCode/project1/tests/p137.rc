

structdef REC2 {
  int a;
};

structdef REC {
  float a;
  float b;
  bool c [32];
  //1 error
  REC2 *x;
  REC2 *x1;
};

REC myRec1;
REC myRec2;

function: void foo1(bool & a[32]) {}

function: void foo2(bool & a[31]) {}

function: void foo3(REC2 & b) {}

function: void main() {
  myRec2 = myRec1;
  myRec2.b = 3.6;
  myRec2.a = myRec2.b;
  
  int x;

  
  foo1(myRec1.c);
  foo3(myRec1.x1[3]);
  
  //3 errors
  foo2(myRec2.c);  
  foo3(myRec1.x1);
  x.b = 3;

}