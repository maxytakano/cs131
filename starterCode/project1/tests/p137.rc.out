Error, "/tmp/cs131_testrunner_OzTaux/file.rc", line 37: 
  Argument of type bool[32] not equivalent to reference parameter a, of type bool[31].
Error, "/tmp/cs131_testrunner_OzTaux/file.rc", line 38: 
  Argument of type REC2* not equivalent to reference parameter b, of type REC2.
Error, "/tmp/cs131_testrunner_OzTaux/file.rc", line 39: 
  Type of expression referenced by "." (int) is not a struct.
Compile: failure.
